examples/combined_ODNP.py

```diff
@@ -133,19 +139,21 @@
 control_thermal.set_prop("acq_params", config_dict.asdict())
 control_thermal.name("control_thermal")
 nodename = control_thermal.name()
-# I don't understand the point of this change -- please explain
+# {{{ on first write, if we can't access the directory, write to a temp file
 try:
-    control_thermal.hdf5_write(f"{filename}", directory=target_directory)
+    control_thermal.hdf5_write(filename, directory=target_directory)
 except:
-    print(
+    final_log.append(
         f"I had problems writing to the correct file {filename}, so I'm going to try to save your file to temp_ctrl.h5 in the current directory"
     )
     if os.path.exists("temp_ctrl.h5"):
-        print("There is already a temp_ctrl.h5 -- I'm removing it")
+        final_log.append("There is already a temp_ctrl.h5 -- I'm removing it")
         os.remove("temp_ctrl.h5")
-        DNP_data.hdf5_write("temp_ctrl.h5", directory=target_directory)
+        target_directory = os.path.getcwd()
```

Comment:
if you're saving to a fallback, it's probably because you have a network problem or the directory is bad, so it should be to a local file.